### ISSUE TITLE
Training small object detection on high-resolution images

### ISSUE THREAD

**Issue by Soda-Wong** [AlexeyAB/darknet#2257]
hello! @AlexeyAB :
     I want to detect a lot of small objects on high-resolution images, but the number of images is just 300. Should I adjust the learning rate and number of iterations? I would like to get a high recall.
    objects:100x100
    images:2600x2000

    I am looking forward to your reply！

**Comment by AlexeyAB** [comment]
@Soda-Wong Hi,
`(100/2600)*608 = 23` it is not very small objects.

So you can try to use: yolov3-tiny.cfg or yolov3.cfg with (`width=608 height=608` in cfg-file), **and add parameter `max=300` for each `[yolo]` layer in cfg-file.**

To increase recall, just just `-thresh 0.1` during detection, it will increase Recall and TP (decrease FN), but it will increase FP too.

Also you can try to use: yolov3-tiny_3l.cfg or yolov3_5l.cfg, if you have objects smaller than 100x100.

**Comment by Soda-Wong** [comment]
@AlexeyAB thanks! I will try!
I am confused whether I need to reduce the number of iterations when my dataset is small.

**Comment by AlexeyAB** [comment]
> I am confused whether I need to reduce the number of iterations when my dataset is small. And Is the usage method of yolov3-tiny_3l.cfg the same as yolov3-voc,cfg?

Just train with `-map` flag and see when mAP will stop.

Usage of yolov3-tiny_3l.cfg is the same as yolov3-voc,cfg

**Comment by zhaoyuanGItHub** [comment]
@AlexeyAB hi, thank your amazing wok! I have  get anchors of my own dataset using the kmeans.py in [https://github.com/qqwweee/keras-yolo3/blob/master/kmeans.py] , but the anchor size is relative to original images, how to change the anchor size relative to original images to size relative to model size.  Is it ok just resizing?

**Comment by AlexeyAB** [comment]
@zhaoyuanGItHub 
Use this command to calculate anchors:
`./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 416 -height 416`
where is 416x416 is width and height from your cfg-file.

**Comment by Soda-Wong** [comment]
@AlexeyAB thank you for your advice!
I use yolov3-tiny_3l.cfg for training. When I trained about 10,000 times, my loss dropped to 6, and now it is almost 30,000 times. It has been stable at 6 and has not continued to decline. How can I adjust it?

there is my learning rate and ites：
`learning_rate=0.001
burn_in=1000
max_batches = 200000
policy=steps
steps=180000,190000
scales=.1,.1`

**Comment by Soda-Wong** [comment]
@AlexeyAB and when the question is How to improve object detection:
you said that:
`for training with a large number of objects in each image, add the parameter max=200 or higher value in the last [yolo]-layer or [region]-layer in your cfg-file (the global maximum number of objects that can be detected by YoloV3 is 0,0615234375*(width*height) where are width and height are parameters from [net] section in cfg-file)`
but I can't find the parameter max, is there something I misunderstood?
